src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -73,6 +73,244 @@ public enum DocumentFlags
+        /// </exception>
+        public TimeSpan? Heartbeat
+        {
+            get => Options.Heartbeat;
+            init => Options.Heartbeat = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the Max number of retry attempts. The retry attempts will reset
+        /// after the replicator is connected to a remote peer. 
+        /// The default is <c>0</c> (<c>10</c> for a single shot replicator or 
+        /// <see cref="int.MaxValue" /> for a continuous replicator is applied.)
+        /// * <c>10</c> for a single shot replicator or <see cref="int.MaxValue" /> for a 
+        /// continuous replicator is applied when user set MaxAttempts to 0.
+        /// * 0 will be returned when default <c>10</c> for a single shot replicator or 
+        /// <see cref="int.MaxValue" /> for a continuous replicator is applied.
+        /// * Setting the value to 1 means that the replicator will try connect once and 
+        /// the replicator will stop if there is a transient error.
+        /// </summary>
+        /// <exception cref="ArgumentException">
+        /// Throw if set the MaxAttempts to a negative value.
+        /// </exception>
+        public int MaxAttempts
+        {
+            get => Options.MaxAttempts;
+            init => Options.MaxAttempts = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the Max delay between retries.
+        /// The default is null (5 min interval is applied).
+        /// * <c>5</c> min interval is applied when MaxAttemptsWaitTime is set to null.
+        /// * null will be returned when default <c>5</c> min interval is applied.
+        /// </summary>
+        /// <exception cref="ArgumentException"> 
+        /// Throw if set the MaxRetryWaitTime to less than 0 full seconds.
+        /// </exception>
+        public TimeSpan? MaxAttemptsWaitTime
+        {
+            get => Options.MaxAttemptsWaitTime;
+            init => Options.MaxAttemptsWaitTime = value;
+        }
+
+        /// <summary>
+        /// Gets the target to replicate with (either <see cref="Database"/>
+        /// or <see cref="Uri"/>
+        /// </summary>
+        [NotNull]
+        public IEndpoint Target { get; }
+
+        /// <summary>
+        /// The implemented custom conflict resolver object can be registered to the replicator 
+        /// at ConflictResolver property. The default value of the conflictResolver is null. 
+        /// When the value is null, the default conflict resolution will be applied.
+        /// </summary>
+        [CanBeNull]
+        public IConflictResolver ConflictResolver { get; init; }
+
+        #if COUCHBASE_ENTERPRISE
+        /// <summary>
+        /// Get or set the way that the replicator will validate TLS certificates.  This
+        /// property will be overriden if the <see cref="PinnedServerCertificate"/> property
+        /// is set.
+        /// </summary>
+        public bool AcceptOnlySelfSignedServerCertificate
```

Comment:
Why has this moved here instead of staying in the EE repo?

Reply:
Actually record can be partial as well.. I can do either way. Maybe I should make it consistent as class?

Reply:
It will make the diff on the EE side easier to read if it remains there, plus the CE repo won't be cluttered with more `#if`